test/jdk/java/net/httpclient/SpecialHeadersTest.java

```diff
@@ -140,6 +157,142 @@ public Object[][] variants() {
     }
 
     static final int ITERATION_COUNT = 3; // checks upgrade and re-use
+    // a shared executor helps reduce the amount of threads created by the test
+    static final TestExecutor executor = new TestExecutor(Executors.newCachedThreadPool());
+    static final ConcurrentMap<String, Throwable> FAILURES = new ConcurrentHashMap<>();
+    static volatile boolean tasksFailed;
+    static final AtomicLong serverCount = new AtomicLong();
+    static final AtomicLong clientCount = new AtomicLong();
+    static final long start = System.nanoTime();
+    public static String now() {
+        long now = System.nanoTime() - start;
+        long secs = now / 1000_000_000;
+        long mill = (now % 1000_000_000) / 1000_000;
+        long nan = now % 1000_000;
+        return String.format("[%d s, %d ms, %d ns] ", secs, mill, nan);
+    }
+
+    final ReferenceTracker TRACKER = ReferenceTracker.INSTANCE;
+    private volatile HttpClient sharedClient;
+
+    static class TestExecutor implements Executor {
+        final AtomicLong tasks = new AtomicLong();
+        Executor executor;
+        TestExecutor(Executor executor) {
+            this.executor = executor;
+        }
+
+        @Override
+        public void execute(Runnable command) {
+            long id = tasks.incrementAndGet();
+            executor.execute(() -> {
+                try {
+                    command.run();
+                } catch (Throwable t) {
+                    tasksFailed = true;
+                    out.printf(now() + "Task %s failed: %s%n", id, t);
+                    err.printf(now() + "Task %s failed: %s%n", id, t);
+                    FAILURES.putIfAbsent("Task " + id, t);
```

Comment:
Hello Daniel, is the `putIfAbsent` intentional here? I wouldn't expect the task id to repeat here (or would there be that many tasks that the `long` id value overflows and repeats?).

Reply:
It's copy-pasted from other tests that use the same technique. Mainly we want to make sure we get the first error. I agree that `putIfAbsent` might not been necessary but it doesn't hurt either: it makes it clear that the value associated with the key will not be overridden.